### PR TITLE
fix: when hit code entry button we send a code email 

### PR DESCRIPTION
**Pull request by danielgent**
If this way isn't immediately clear I could create a new route to do this :see_no_evil:

### CODE REVIEW

File: src/routes/tsx/routes.tsx [hunk 1]
```diff
@@ -682,6 +682,12 @@ export const loginRoutes = new OpenAPIHono<{ Bindings: Env; Variables: Var }>()
             "application/x-www-form-urlencoded": {
               schema: z.object({
                 username: z.string().transform((u) => u.toLowerCase()),
+                login_selection: z
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:685]: :eyes: All will be explained below :smile:

File: src/routes/tsx/routes.tsx [hunk 2]
```diff
@@ -729,15 +735,18 @@ export const loginRoutes = new OpenAPIHono<{ Bindings: Env; Variables: Var }>()
       session.authParams.username = params.username;
       await env.data.universalLoginSessions.update(session.id, session);
 
-      const passwordLoginSelection =
-        parsePasswordLoginSelectionCookie(
-          getCookie(ctx, getPasswordLoginSelectionCookieName(client.id)),
-        ) || SesamyPasswordLoginSelection.code;
+      // we want to be able to override this with a value in the POST
+      if (params.login_selection !== SesamyPasswordLoginSelection.code) {
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:739]: Auth0 seems to do this kind of thing - accepting different params in the post body to do different things

File: src/utils/components/LoginPage.tsx [hunk 3]
```diff
@@ -71,14 +71,13 @@ const LoginPage: FC<Props> = ({ error, vendorSettings, email, state }) => {
...
           </div>
-          <Button
-            Component="a"
-            href={`/u/enter-code?${loginLinkParams.toString()}`}
-            variant="secondary"
-            className="block"
-          >
-            {i18next.t("enter_a_code_btn")}
-          </Button>
+          <form method="post" action={`/u/code?${loginLinkParams.toString()}`}>
```
Comment (danielgent) [inline src/utils/components/LoginPage.tsx:74]: :eyes: I was thinking of how to do this for a while...

This form is used on `/u/login` so here I'm posting to a different URL

File: src/utils/components/LoginPage.tsx [hunk 4]
```diff
@@ -71,14 +71,13 @@ const LoginPage: FC<Props> = ({ error, vendorSettings, email, state }) => {
...
+            <input type="hidden" name="login_selection" value="code" />
```
Comment (danielgent) [inline src/utils/components/LoginPage.tsx:75]: By doing this, I'm forcing this endpoint to then redirect to the enter code step and *send a code email*

Comments welcome if this is too cryptic. Failing this I could make another endpoint, which could be an anchor and send an email on a get (but I think a `<button/>` inside a form is more semantic as gets shouldn't really do things like send emails... I think... idempotent? :smile:

File: src/routes/tsx/routes.tsx (hunk 1 above)
Comment (markusahlstrand) [reply src/routes/tsx/routes.tsx:685]: I think this would be the same a `z.enum(["a","b"])` right?

**Review by markusahlstrand — APPROVED** [review]
Looks good. Maybe we can use a z.enum with a default? Then it's clear what is the default behavior

**Pull request merged**